xtask/src/main.rs

```diff
@@ -0,0 +1,34 @@
+use clap::{Parser, Subcommand};
+
+mod publish;
+mod runchecks;
+
+#[derive(Parser)]
+#[command(author, version, about, long_about = None)]
+struct Args {
+    #[command(subcommand)]
+    command: Command,
+}
+
+#[derive(Subcommand)]
+enum Command {
+    /// Publish a crate to crates.io
+    Publish {
+        /// The name of the crate to publish on crates.io
+        name: String,
+    },
+    /// Run the specified `burn` tests and checks locally.
+    RunChecks {
+        /// The environment to run checks against
+        env: runchecks::CheckType,
+    },
+}
+
+fn main() -> anyhow::Result<()> {
+    let args = Args::parse();
+
+    match args.command {
+        Command::RunChecks { env } => runchecks::run(env),
+        Command::Publish { name } => publish::run(name),
```

Comment:
Can we unwrap the `run` functions here instead of returning an error?

Reply:
I would recommend against doing this, as unwrapping here makes it appear to users that `cargo-xtask` had encountered an internal error.
It's best to return the result so that the error gets printed and the program exits with a normal error status code.

Reply:
I agree with this analysis